src/corehost/cli/hostpolicy.cpp

```diff
@@ -615,6 +614,54 @@ SHARED_API int corehost_get_com_activation_delegate(void **delegate)
         delegate);
 }
 
+SHARED_API int corehost_get_load_and_execute_in_memory_assembly_delegate(void** delegate)
```

Comment:
I think I agree now. Basically what we want is a way to ask for a special "runtime" entry point. And we have 3 of them now (will have 4 soon). I agree it would be nice to have just one export for all of them - it would also make it a bit more explicit ... "these are special things, please don't use them directly".

Reply:
I've updated the code to use an enum in the layer between hostfxr and hostpolicy. Do we want to use an enum as well on the public API surface of hostfxr? I think we can do that as well if we want to.

Reply:
Yes - I think we should do the same change in hostfxr as well.